src/app/test-record/test-record.component.spec.ts

```diff
@@ -1,55 +1,115 @@
-import {async, ComponentFixture, getTestBed, TestBed} from '@angular/core/testing';
-
-import {TestRecordComponent} from './test-record.component';
-import {INITIAL_STATE, Store} from '@ngrx/store';
-import {IAppState} from '@app/store/state/app.state';
-import {hot} from 'jasmine-marbles';
-import {ActivatedRoute, convertToParamMap} from '@angular/router';
-import {Observable, of} from 'rxjs';
+import { async, ComponentFixture, getTestBed, TestBed } from '@angular/core/testing';
+import { TestRecordComponent } from './test-record.component';
+import { INITIAL_STATE, Store } from '@ngrx/store';
+import { IAppState } from '@app/store/state/app.state';
+import { hot } from 'jasmine-marbles';
+import { SharedModule } from '@app/shared/shared.module';
+import { TestType } from '@app/models/test.type';
+import { TestResultModel } from '@app/models/test-result.model';
+import { CUSTOM_ELEMENTS_SCHEMA, SimpleChange, SimpleChanges } from '@angular/core';
+import { TestRecordTestType } from '@app/models/test-record-test-type';
+import { FormGroup, FormGroupDirective, ReactiveFormsModule } from '@angular/forms';
+import { MatDialog, MatDialogModule } from '@angular/material/dialog';
+import { TestRecordMapper, TestTypesApplicable } from '@app/test-record/test-record.mapper';
+import { VIEW_STATE } from '@app/app.enums';
+import { Preparer } from '@app/models/preparer';
+import { TestStation } from '@app/models/test-station';
+import { Location } from '@angular/common';
+import { SpyLocation } from '@angular/common/testing';
 
 describe('TestRecordComponent', () => {
   let component: TestRecordComponent;
   let fixture: ComponentFixture<TestRecordComponent>;
   let store: Store<IAppState>;
   let injector: TestBed;
+  let location;
+  let changes: SimpleChanges;
+  let dialog: MatDialog;
+  let testRecordMapper;
+  const testRecord = { vehicleType: 'psv' } as TestResultModel;
+  const testType = {} as TestType;
+  const testObject: TestRecordTestType = { testRecord: testRecord, testType: testType };
+  const testResultParentForm = new FormGroup({ testType: new FormGroup({}) });
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
+      imports: [SharedModule, ReactiveFormsModule, MatDialogModule],
       declarations: [TestRecordComponent],
+      schemas: [CUSTOM_ELEMENTS_SCHEMA],
       providers: [
+        TestRecordMapper,
+        FormGroupDirective,
         {
           provide: Store,
           useValue: {
             dispatch: jest.fn(),
-            pipe: jest.fn(() => hot('-a', {a: INITIAL_STATE})),
+            pipe: jest.fn(() => hot('-a', { a: INITIAL_STATE })),
```

Comment:
we should remove store references in the component

Reply:
I tried removing as we already talked about this and the test was failing